frontends/election-manager/src/components/election_manager_tally_report.tsx

```diff
@@ -57,16 +61,33 @@ export function ElectionManagerTallyReport({
   fullElectionTally,
   generatedAtTime = new Date(),
   tallyReportType,
-  partyId: partyIdFromProps,
+  partyId: reportPartyIdFromProps,
   precinctId: precinctIdFromProps,
   scannerId,
   votingMethod,
   officialCandidateWriteIns,
 }: Props): JSX.Element {
-  const ballotStylePartyIds =
-    partyIdFromProps !== undefined
-      ? [unsafeParse(PartyIdSchema, partyIdFromProps)]
-      : Array.from(new Set(election.ballotStyles.map((bs) => bs.partyId)));
+  const reportPartyId: Optional<PartyId> = reportPartyIdFromProps
+    ? unsafeParse(PartyIdSchema, reportPartyIdFromProps)
+    : undefined;
+
+  // The report party id represents the party of the entire report. If it
+  // exists, the report only contains votes from ballots associated with that
+  // party. The section party id represents the party of an individual section
+  // (usually a a page) within the report.
+  const sectionPartyIds = (() => {
+    // If the report is not specific to a party, include page for each party
+    // including a page for nonpartisan races.
+    if (!reportPartyId) {
+      return getPartyIdsWithContests(election);
```

Comment:
I struggled with a name for this helper, but here's the JSDoc: 
```
/**
 * Returns the ids of all parties with an associated contest defined. Will
 * return `undefined` for nonpartisan contests.
 */
```
So it will include `undefined`. Suggestions on a clearer name? Maybe `getContestPartyIds` is marginally clearer?

Reply:
ooh I see, I think I was just looking at some other function implementation and then got confused. I think the name is fine.